modules/imap/hm-ews.php

```diff
@@ -681,9 +695,11 @@ public function message_action($action, $itemIds, $folder=false, $keyword=false)
     }
 
     public function get_message_headers($itemId) {
+        $base64Id = $this->get_item_id_base64($itemId);
         $request = array(
             'ItemShape' => array(
                 'BaseShape' => 'AllProperties',
+                'IncludeMimeContent' => true,
```

Comment:
This will be a huge effort on the server and the network to return the full mime content every time with the headers. Why do you add it here? It should be part of get message content only.